src/sharding/Shard.js

```diff
@@ -117,11 +125,20 @@ class Shard extends EventEmitter {
           execArgv: this.execArgv,
         })
         .on('message', this._handleMessage.bind(this))
-        .on('exit', this._exitListener);
+        .on('exit', this._exitListener)
+        .on('disconnect', this._handleDisconnect.bind({ parentProcessDisconnect: true }));
```

Comment:
```suggestion
        .on('disconnect', this._handleDisconnect.bind(this, { parentProcessDisconnect: true }));
```

Reply:
slight oversight on my part, thanks for the suggestion